src/pointtopoint.jl

```diff
@@ -204,13 +212,19 @@ $(_doc_external("MPI_Irecv"))
 """
 Irecv!(recvbuf, comm::Comm; source::Integer=Consts.MPI_ANY_SOURCE[], tag::Integer=Consts.MPI_ANY_TAG[]) =
     Irecv!(recvbuf, source, tag, comm)
-function Irecv!(buf::Buffer, source::Integer, tag::Integer, comm::Comm)
-    req = Request()
+function Irecv!(data::Union{MPIPtr, MPIBuffertype}, count::Integer, datatype::Union{MPI_Datatype, Datatype}, source::Integer, tag::Integer, comm::Union{MPI_Comm, Comm}, req::Union{MPI_Request, Request})
```

Comment:
It doesn't make sense to include `MPI_Request` here in the signature, as it is an immutable object so can't be modified (you could use `Ptr{MPI_Request}`?)